### PR TITLE
 doc: refactor `hashes` group

### PR DESCRIPTION
**Pull request by jia200x**
### Contribution description
Following #9842, this PR groups all hashes into the `hashes` group. I also define/modify some other groups in order to create the categories Checksums, Keyed cryptographic hash functions, Unkeyed cryptographic hash functions and Non cryptographic hash functions (all of them under `sys_hashes` group).

This PR does the following:

- Checksums moved from `sys`group to `sys_hashes` group
- Define groups for `sha256` and `sha3`
- Add CMAC to `sys_hashes_keyed`
- Add MD5, SHA1, SHA256 and SHA3 to `sys_hashes_unkeyed`
- Add all hashes from `hashes.h` files to `sys_hashes_non_crypto`

I also deprecated the section based documentation from the `doc.txt`, because it was outdated and it had the same information as the Doxygen group list.

### Testing procedure
`make doc`. Go to `System>Hashes` in the generated documentation and see how these components are grouped.

### Issues/PRs references
#8479

### CODE REVIEW

File: sys/checksum/doc.txt [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 /**
  * @defgroup    sys_checksum Checksum
- * @ingroup     sys
+ * @ingroup     sys_hashes
```
Comment (PeterKietzmann) [inline sys/checksum/doc.txt]: Why do you put checksum in `sys_hashes`? I would guess that (as long as we keep both checksum/ and hashes/ in separate folders) we would keep both `@ingroup sys` ?

Comment (jia200x) [reply sys/checksum/doc.txt]: As I have read, strictly speaking checksums are hashes. See for instance [this entry](https://en.wikipedia.org/wiki/List_of_hash_functions).
Although they are in different folder, I'm trying to organize them by topic :)

**Comment by jia200x** [comment]
> Nice! Did you also have a look at the header files in sys/include/checksum/?

Yes, since I move the `sys_checksum` group to `sys_hashes`, these headers are not affected by the change (the `@ingroup` directive is in charge of moving the group)

File: sys/checksum/doc.txt (hunk 1 above)
Comment (PeterKietzmann) [reply sys/checksum/doc.txt]: I was referring to the mismatch between folder structure and documentation. But I agree that it looks way nicer this way. So the conclusion might be to consider file shifting which is not the purpose of this PR and maybe not wanted anyway.

**Comment by PeterKietzmann** [comment]
>  these headers are not affected by the change

Sure, I was more thinking about the need for further cleanup. Otherwise I'm fine with this PR.

File: sys/checksum/doc.txt (hunk 1 above)
Comment (jia200x) [reply sys/checksum/doc.txt]: Ah, I see. Yes, could be an option. I would be in for moving the folder in another PR

**Comment by jia200x** [comment]
> Sure, I was more thinking about the need for further cleanup. Otherwise I'm fine with this PR.

Good! :)

**Comment by PeterKietzmann** [comment]
PS: I'm wondering if we need that many commits for this kind of change

**Comment by jia200x** [comment]
> PS: I'm wondering if we need that many commits for this kind of change

Maybe it doesn't make that much sense. I will try to commit by group rather than by feature.

**Pull request merged**